psyneulink/core/llvm/helpers.py

```diff
@@ -668,4 +669,105 @@ def generate_sched_condition(self, builder, condition, cond_ptr, node,
 
             return run_cond
 
+        elif isinstance(condition, Threshold):
+            def get_single_element_depth(val):
+                cur_val = val.type.pointee
+                depth = 1
+
+                # always check the first dimension for multiple values
+                while True:
+                    if cur_val.count != 1:
+                        return None
+
+                    if (
+                        hasattr(cur_val, 'element')
+                        and isinstance(cur_val.element, ir.ArrayType)
+                    ):
+                        cur_val = cur_val.element
+                        depth += 1
+                    else:
+                        break
+
+                return depth
+
+            target = condition.dependency
+            param = condition.parameter
+            threshold = condition.threshold
+            comparator = condition.comparator
+            indices = condition.indices
+
+            tolerance_value = condition.rtol * abs(threshold) + condition.atol
```

Comment:
It could replace some of the checks. I think a manual value would still need to be used for the strict inequalities, though I'm not really sure how much a tolerance even makes sense for those.

Reply:
agreed, I think the semantics need a bit of discussion.
using is_close in all of them made sense to me because it preserves the expected comparison invariants,
but at the same time breaking 'x + epsilon > x' is not very intuitive either.
I'm not sure what the user expectations are here, and how much we can restrict them.
Allowing only shape inequalities can be one way out, users can always combine scheduling decisions themselves.